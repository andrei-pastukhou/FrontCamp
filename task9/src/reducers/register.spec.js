import reducer from './register'
const initialState = {
    message: '',
    pending: false,
}

describe('register reducer', () => {
    it('should handle initial state', () => {
        expect(
            reducer(undefined, {})
        ).toEqual(initialState)
    });
    it('should handle REGISTER', () => {
        expect(
            reducer(initialState, {
                type: 'REGISTER',
            })
        ).toEqual({
            message: '',
            pending: true,
        })
    });
    it('should handle REGISTER_SUCCSESS', () => {
        expect(
            reducer(initialState, {
                type: 'REGISTER_SUCCSESS',
                message: 'test message'
            })
        ).toEqual({
            pending: false,
            message: 'test message'
        })
    });
    it('should handle REGISTER_ERROR', () => {
        expect(
            reducer(initialState, {
                type: 'REGISTER_ERROR',
                message: 'test error message'
            })
        ).toEqual({
            pending: false,
            message: 'test error message'
        })
    });
});


describe('register reduxer with snapshots ', () => {
    it('should handle initial state', () => {
        const action = {}
        expect(reducer(undefined, action)).toMatchSnapshot();
    });
    it('should handle REGISTER', () => {
        const action = {
            type: 'REGISTER',
        }
        expect(reducer(initialState, action)).toMatchSnapshot();
    });
    it('should handle REGISTER_SUCCSESS', () => {
        const action = {
            type: 'REGISTER_SUCCSESS',
            message: 'test message'
        }
        expect(reducer(initialState, action)).toMatchSnapshot();
    });
    it('should handle REGISTER_ERROR', () => {
        const action = {
            type: 'REGISTER_ERROR',
            message: 'test error message'
        }
        expect(reducer(initialState, action)).toMatchSnapshot();
    });
});