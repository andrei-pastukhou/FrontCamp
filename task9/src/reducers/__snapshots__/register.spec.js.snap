// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register reduxer with snapshots  should handle REGISTER 1`] = `
Object {
  "message": "",
  "pending": true,
}
`;

exports[`register reduxer with snapshots  should handle REGISTER_ERROR 1`] = `
Object {
  "message": "test error message",
  "pending": false,
}
`;

exports[`register reduxer with snapshots  should handle REGISTER_SUCCSESS 1`] = `
Object {
  "message": "test message",
  "pending": false,
}
`;

exports[`register reduxer with snapshots  should handle initial state 1`] = `
Object {
  "message": "",
  "pending": false,
}
`;
